### ISSUE TITLE
Handling OOM  during the optimizer.step() phase

### ISSUE THREAD

**Issue by ememos** [ai-computing/aicomp#18]
An OOM  error might occur in GPU memory during the execution of optimizer.step()

**Comment by ememos** [comment]
The model_offload option added. With this option, during the optimizer.step() interval, parameters, gradients, and optimizer states are all offloaded to the CPU, and the optimizer step is performed on the CPU.

**Comment by ememos** [comment]
The option name has been changed to swap_model_in_optstep.
